orchestrator/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,481 @@
+	return nil
+}
+
+func (k *Kubernetes) updateCurrentNode() error {
+	var err error
+	node := &types.NodeInfo{
+		IP:               k.IP,
+		OrchestratorPort: types.DefaultOrchestratorPort,
+	}
+	node.Name, err = os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	uuid, err := ioutil.ReadFile(hostUUIDFile)
+	if err == nil {
+		node.ID = string(uuid)
+		k.currentNode = node
+		return nil
+	}
+
+	// file doesn't exists, generate new UUID for the host
+	node.ID = util.UUID()
+	if err := os.MkdirAll(cfgDirectory, os.ModeDir|0600); err != nil {
+		return fmt.Errorf("Fail to create configuration directory: %v", err)
+	}
+	if err := ioutil.WriteFile(hostUUIDFile, []byte(node.ID), 0600); err != nil {
+		return fmt.Errorf("Fail to write host uuid file: %v", err)
+	}
+
+	k.currentNode = node
+	return nil
+}
+
+func (k *Kubernetes) GetCurrentNode() *types.NodeInfo {
+	return k.currentNode
+}
+
+func (k *Kubernetes) CreateController(req *orchestrator.Request) (instance *orchestrator.Instance, err error) {
+	defer func() {
+		err = errors.Wrapf(err, "fail to create controller for %v", req.VolumeName)
+	}()
+
+	if err := orchestrator.ValidateRequestCreateController(req); err != nil {
+		return nil, err
+	}
+
+	if req.NodeID != k.currentNode.ID {
+		return nil, fmt.Errorf("incorrect node, requested %v, current %v", req.NodeID,
+			k.currentNode.ID)
+	}
+
+	cmd := []string{
+		"launch", "controller",
+		"--listen", "0.0.0.0:9501",
+		"--frontend", "tgt",
+	}
+	for _, url := range req.ReplicaURLs {
+		waitURL := strings.Replace(url, "tcp://", "http://", 1) + "/v1"
+		if err := util.WaitForAPI(waitURL, WaitAPITimeout); err != nil {
+			return nil, err
+		}
+		cmd = append(cmd, "--replica", url)
+	}
+	cmd = append(cmd, req.VolumeName)
+
+	podNode, err := k.getCurrentNodePod()
+	if err != nil {
+		logrus.Errorf("fail to get daemonset pod info")
+		return nil, err
+	}
+
+	privilege := true
+	pod := &apiv1.Pod{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: req.InstanceName,
+		},
+		Spec: apiv1.PodSpec{
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": podNode.Spec.NodeName,
+			},
+			Containers: []apiv1.Container{
+				{
+					Name:    req.InstanceName,
+					Image:   k.EngineImage,
+					Command: cmd,
+					SecurityContext: &apiv1.SecurityContext{
+						Privileged: &privilege,
+					},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "dev",
+							MountPath: "/host/dev",
+						},
+						{
+							Name:      "proc",
+							MountPath: "/host/proc",
+						},
+					},
+				},
+			},
+			Volumes: []apiv1.Volume{
+				{
+					Name: "dev",
+					VolumeSource: apiv1.VolumeSource{
+						HostPath: &apiv1.HostPathVolumeSource{
+							Path: "/dev",
+						},
+					},
+				},
+				{
+					Name: "proc",
+					VolumeSource: apiv1.VolumeSource{
+						HostPath: &apiv1.HostPathVolumeSource{
+							Path: "/proc",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podController, err := k.cli.CoreV1().Pods(apiv1.NamespaceDefault).Create(pod)
```

Comment:
We cannot always do `apiv1.NamespaceDefault`. We should deploy to the namespace user wanted. Just detect which namespace current manager is running on for now.